pytorch_lightning/callbacks/base.py

```diff
@@ -46,6 +46,30 @@ def on_sanity_check_end(self, trainer, pl_module):
         """Called when the validation sanity check ends."""
         pass
 
+    def on_train_epoch_start(self, trainer, pl_module):
```

Comment:
rather, the `on_train_epoch_start` invokes that there are separate epochs for train/valid/test
```suggestion
    def on_epoch_train_start(self, trainer, pl_module):
```

Reply:
there are